ramiray/c01 git02

ramiray opened the pull request:
# Description

Exercise (e.g. C01-E01): 

Student Name:

> A brief explanation of what has been done.

## Type of change

- [ ] Exercise Submission (one exercise per PR)
  - [ ] This PR is just for **one** exercise of a class
  - [ ] PR name follows the pattern `<github-username>/<exercise-number>`
  - [ ] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

denstorti commented on classes/01class/exercises/c01-git02/ramiray/Readme.md:
````diff
@@ -0,0 +1,27 @@
+yes we can .
+using the following command : 
+```
+ git reflog 
````
what if you don't have `git reflog`, e.g., you just cloned the repo?

ramiray replied:
Hi Denis , I had already created separate PR for both   c01-aws02 and c01-git01as well as other exercises. I'm bit confused which you are referring to.

ramiray replied:
Hi Denis,  Is it git log oneline  command. 

As per my understanding both the commands(reflog and log oneline) list out all the commits that we do . Only difference is that the reflog shows all the commit and commit discarded and more specific . Please clarify if I'm  wrong. 

Thanks,
Ramya.

roperto replied:
Hi @ramiray 

You are right, when you use `git reflog` you can see all the discarded commits. The correct term is "dangling" or "unreachable" commits.

When you squash commits, all the "squashed" commits are removed from the history and replaced by a single commit with all the content of those deleted commits. If no other branches have them, them become "dangling" (discarded).

A few different things could trigger a "cleanup" that would permanently delete those dangling commits, so it is not safe to assume they will be there forever.
 
Also, if you pull a squashed commit from GitHub, it will not bring all those discarded commits -- so in that case, is it possible to see the "inner commits" ?

roperto commented:
Hi @ramiray 

Same thing in this one, this is a pull request for c01-git02 but you have together the answers for two other exercises

![image](https://user-images.githubusercontent.com/3672494/86586393-2946e600-bfcb-11ea-8bf4-137f69556e45.png)

What you did is probably starting you new exercise using the branch of the other exercise, which believe me is a very common mistake.

The baseline is: when creating a new branch for the exercise, checkout "master" first again.

If you need some help please reach out, I already helped a few other students with the exact same mistake.

Cheers

ramiray commented:
Hi Daniel,

Yes, I'm a very new starter  to Git and I made a lot of mistakes during
class01 exercises. I hope I didn't make many errors in class02 exercises. I
will do the necessary changes and let you know.


Thanks & Regards,
Ramya



On Mon, Jul 6, 2020 at 8:59 PM Daniel Thee Roperto <notifications@github.com>
wrote:

> Hi @ramiray <https://github.com/ramiray>
>
> Same thing in this one, this is a pull request for c01-git02 but you have
> together the answers for two other exercises
>
> [image: image]
> <https://user-images.githubusercontent.com/3672494/86586393-2946e600-bfcb-11ea-8bf4-137f69556e45.png>
>
> What you did is probably starting you new exercise using the branch of the
> other exercise, which believe me is a very common mistake.
>
> The baseline is: when creating a new branch for the exercise, checkout
> "master" first again.
>
> If you need some help please reach out, I already helped a few other
> students with the exact same mistake.
>
> Cheers
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/devopsacademyau/academy/pull/794#issuecomment-654163979>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJEJB4GO6ITWHUBXHH6KWIDR2GVBVANCNFSM4OFK5BDA>
> .
>

roperto commented:
Don't worry about it ... I use git for a few years and I still do a lot of mistakes :-)

ramiray commented:
Hi Daniel,

I think we can achieve this by git rebase -i HEAD~<number to which we need
to commit> and squash those commits.  All the inner commits are
squashed/merged as a single one.  Please correct me if I'm wrong.

Thanks ,
Ramya.



On Mon, Jul 6, 2020 at 10:49 PM Daniel Thee Roperto <
notifications@github.com> wrote:

> *@roperto* requested changes on this pull request.
>
> The wrong file has been removed, thanks.
>
> Just continuing the conversation from Denis about reflog (see comments
> above/below)
> ------------------------------
>
> In classes/01class/exercises/c01-git02/ramiray/Readme.md
> <https://github.com/devopsacademyau/academy/pull/794#discussion_r450196493>
> :
>
> > @@ -0,0 +1,27 @@
> +c01-git02
> +
> +1. Is it possible to see the inner commit of squashed commit with git rebase?
> +yes we can .
> +using the following command :
> +```
> + git reflog
>
> Hi @ramiray <https://github.com/ramiray>
>
> You are right, when you use git reflog you can see all the discarded
> commits. The correct term is "dangling" or "unreachable" commits.
>
> When you squash commits, all the "squashed" commits are removed from the
> history and replaced by a single commit with all the content of those
> deleted commits. If no other branches have them, them become "dangling"
> (discarded).
>
> A few different things could trigger a "cleanup" that would permanently
> delete those dangling commits, so it is not safe to assume they will be
> there forever.
>
> Also, if you pull a squashed commit from GitHub, it will not bring all
> those discarded commits -- so in that case, is it possible to see the
> "inner commits" ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/devopsacademyau/academy/pull/794#pullrequestreview-443043348>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJEJB4EU7IOE3BSEEBA6NU3R2HB4ZANCNFSM4OFK5BDA>
> .
>
